src/commonMain/kotlin/exchange.dydx.abacus/calculator/MarginModeCalculator.kt

```diff
@@ -15,6 +16,18 @@ object MarginModeCalculator {
     ): MutableMap<String, Any>? {
         val modified = tradeInput?.mutable() ?: return null
         val marketId = parser.asString(tradeInput["marketId"])
+        val market = parser.asMap(markets?.get(marketId))
+
+        val imf = parser.asDouble(parser.value(market, "configs.initialMarginFraction")) ?: Numeric.double.ZERO
+        val effectiveImf = parser.asDouble(parser.value(market, "configs.effectiveInitialMarginFraction")) ?: Numeric.double.ZERO
+        val maxMarketLeverage = if (effectiveImf > Numeric.double.ZERO) {
+            Numeric.double.ONE / effectiveImf
+        } else if (imf > Numeric.double.ZERO) {
+            Numeric.double.ONE / imf
+        } else {
+            Numeric.double.ONE
+        }
```

Comment:
the calculation for `maxMarketLeverage` is duplicated in quite a few places. Would it be easier to have a pure function in `marketCalculator` or something similar and just use in each of these cases?

Reply:
Similar to what Rui said, if it can be included in the actual configs struct it would be easy to de-dupe this calculation